### DISCUSSION TITLE
How does caching work with nuxt/image?

### DISCUSSION THREAD

**Discussion by raphaelbernhart** [nuxt/image#1198] [Q&A]
Hi :)

while developing an application with nuxt3 and nuxt/image (IPX / local) I came upon a Page Speed Insights problem ([image](https://github.com/nuxt/image/assets/48283236/ea60e033-e24b-458d-871c-5d8f19e4e187)).

As far as I know there is no cache busting functionality implemented so far. The default TTL is 60s which is not long enough for me. To at least improve the load speed when navigating on the site a few minutes after the initial request, I increased the TTL to 1h.

How are you guys handling caching with nuxt/image? Just simply changing the resource file names? IMO this is a lot of recurring work when having a lot of local images. I am aware that I could use a third party service like cloudinary but I like the flexibility which comes with locally hosting the images.

I think something like vite is doing (but in the assets folder not the public folder) would be nice, in a perfect world, tho there probably is something which is hindering such functionality.

**Comment by SultanDino** [comment]
bump

**Comment by SultanDino** [comment]
I have done it with a custom provider.

`providers/ipx.ts`

```typescript
import type { ProviderGetImage } from '@nuxt/image';
import { getImage as ipxProvider } from '#image/providers/ipx';

export const validateDomains = true;
export const supportsAlias = true;
export const getImage: ProviderGetImage = (src, options, ctx) => {
	options.modifiers.v = '1.00'; // or you can use env variable defined in nuxt config e.g. useRuntimeConfig().public.IMAGE_REVISION_NUMBER

	return ipxProvider(src, options, ctx);
};
``` 

Then we overwrite the default ipx provider with our custom provider
`nuxt.config.ts`
```typescript
export default defineNuxtConfig({
        // ...
	image: {
		providers: {
			ipx: {
				name: 'ipx',
				provider: '~/providers/ipx.ts',
			},
		},
	},
})
```

Then in your component, you don't need to define the provider attribute, since ipx is already default in nuxt/image.
```
<nuxt-img src="/images/test.jpg" width=150 height=75 />
```

This will output an `img` element with `src`
```
/_ipx/v_1.00&s_150x75/images/test.jpg
```
